firmware/library/L1_Drivers/uart.hpp

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+class UARTInterface
+{
+ public:
+    virtual void setBaud(unsigned int baudrate) = 0;
+    virtual bool init(unsigned int baud, uint32_t queue_size = 16) = 0;
+    virtual void IntHandle() = 0;
```

Comment:
Write the full name of the function and what it is doing. A function or method should say and action and not be a nown. For example, would prefer `AttachInterruptHandler(...)`

Also, the method should accept a function pointer.